From 4567c111029e1a89b0c7c9c990528cbcb96400ce Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 19 Jan 2024 13:13:16 +0100
Subject: [PATCH 0645/1695] dts: stm32: add frequency monitor to
 stm32mp251.dtsi SoC file

Adds HSE frequency monitor to stm32mp251.dtsi SoC file.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 679e89393bb6af78a51b72726bf122d4fc1622ef dts: stm32: add frequency monitor to stm32mp251.dtsi SoC file]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I3de363a47157f739bcf1f893f2b47e66e359354d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362463
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index d41338349..1fec7a4c5 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -26,6 +26,14 @@
 		};
 	};
 
+	hse_monitor: hse-monitor {
+		compatible = "st,freq-monitor";
+		counter = <&lptimer1 1 5 1 1>;
+		clocks = <&rcc HSE_CK>, <&rcc HSI_CK>;
+		clock-names = "hse", "hsi";
+		status = "disabled";
+	};
+
 	psci {
 		compatible = "arm,psci-1.0";
 		method = "smc";
-- 
2.47.2

